static/js/search_suggestion.js

```diff
@@ -238,11 +150,33 @@ function get_group_suggestions(all_people, operators) {
         return {description: description, search_string: search_string};
     });
 
+    if (!match_criteria(operators, [{operator: "is", operand: "private"}])) {
+        suggestions.push({description: "Private messages", search_string:"is:private"});
+    }
     return suggestions;
 }
 
-function get_person_suggestions(all_people, query, autocomplete_operator) {
-    if (query === '') {
+function get_person_suggestions(all_people, last, operators, autocomplete_operator) {
```

Comment:
What are your thoughts on breaking this into two different functions, one for sender and for pm-with?  I think two simple functions will be easier to maintain than one pretty complex function.

If you decide you want to keep the structure, I think we should add some comments for some of the logic here.